lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FieldToPointFunctionalChaosAlgorithm.cxx

```diff
@@ -0,0 +1,451 @@
+  Bool isIndependent = true;
+  for (UnsignedInteger j = 0; j < dimension && isIndependent; ++ j)
+  {
+    const Sample marginalJ = sample.getMarginal(j);
+    for (UnsignedInteger i = j + 1; i < dimension && isIndependent; ++ i)
+    {
+      TestResult testResult(HypothesisTest::Spearman(sample.getMarginal(i), marginalJ));
+      isIndependent = isIndependent && testResult.getBinaryQualityMeasure();
+    }
+  }
+
+  ComposedDistribution distribution(marginals);
+  if (!isIndependent)
+    distribution.setCopula(EmpiricalBernsteinCopula(sample, sample.getSize()));
+  return distribution;
+}
+
+/* Response surface computation */
+void FieldToPointFunctionalChaosAlgorithm::run()
+{
+  const UnsignedInteger size = inputProcessSample_.getSize();
+  Collection<KarhunenLoeveResult> klResultCollection(blockIndices_.getSize());
+  Point eigenValues;
+
+  ProcessSample inputProcessSample(inputProcessSample_);
+  InverseTrendTransform inverseTransform;
+  if (!centeredSample_)
+  {
+    inverseTransform = InverseTrendTransform(P1LagrangeEvaluation(inputProcessSample_.computeMean()), inputProcessSample_.getMesh());
+    inputProcessSample = inverseTransform(inputProcessSample_);
+  }
+
+  for (UnsignedInteger i = 0; i < blockIndices_.getSize(); ++ i)
+  {
+    const ProcessSample inputProcessSampleI(inputProcessSample.getMarginal(blockIndices_[i]));
+    const Bool centered = true;
+    KarhunenLoeveSVDAlgorithm algo(inputProcessSampleI, threshold_, centered);
+    algo.setNbModes(nbModes_);
+    algo.run();
+    eigenValues.add(algo.getResult().getEigenvalues());
+    klResultCollection[i] = algo.getResult();
+    LOGINFO(OSS() << "block #" << i << " indices=" << blockIndices_[i] << " ev size=" << algo.getResult().getEigenvalues().getDimension());
+  }
+
+  if (recompress_)
+  {
+    const Scalar cumulatedVariance = eigenValues.norm1();
+    std::sort(eigenValues.begin(), eigenValues.end(), [](const Scalar a, const Scalar b) { return a > b; });
+    const UnsignedInteger nbModesMax = std::min(nbModes_, eigenValues.getSize());
+    // Find the cut-off in the eigenvalues
+    UnsignedInteger K = 0;
+    Scalar selectedVariance = 0.0; // sum of eigenvalues selected after cut-off is applied
+    do
+    {
+      selectedVariance += eigenValues[K];
+      ++ K;
+    } while ((K < nbModesMax) && (selectedVariance < (1.0 - threshold_) * cumulatedVariance));
+    LOGINFO(OSS() << "Selected " << K << " eigenvalues out of " << eigenValues.getSize() << " computed");
+
+    const Scalar lambdaCut = eigenValues[K];
+    for (UnsignedInteger i = 0; i < blockIndices_.getSize(); ++ i)
+    {
+      UnsignedInteger Ki = 1;// retain at least 1 mode
+      Point eigenValuesI(klResultCollection[i].getEigenvalues());
+      Scalar selectedVarianceI = eigenValuesI[0];
+      const Scalar cumulatedVarianceI = eigenValuesI.norm1();
+      while ((Ki < eigenValuesI.getSize()) && (eigenValuesI[Ki] >= lambdaCut))
```

Comment:
Why `>=` ?

Reply:
so if eigenvalues < lambdaCut, we drop them? I thought that we keep ev <= lambdaCut